### PR TITLE
timg: init at 2020-04-09, tiv: init at 2020-04-09

### PR DESCRIPTION
**Pull request by ToxicFrog**
###### Motivation for this change
These are simple image viewers for the tty, similar to `catimg` but with different feature sets -- `timg` has better scaling options, and `tiv` uses 8x4 block drawing characters to achieve higher resolution.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ToxicFrog** [comment]
Ping?

**Comment by Lassulus** [comment]
`Add myself to maintainer list` is kinda right. but please use your nickname in the commit :)

**Comment by ToxicFrog** [comment]
> `Add myself to maintainer list` is kinda right. but please use your nickname in the commit :)

Done.

File: pkgs/tools/misc/timg/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{ stdenv, pkgs, fetchFromGitHub }:
```
Comment (Lassulus) [inline pkgs/tools/misc/timg/default.nix:1]: although this is possible, its preferred to explicitly import the packages and not the whole pkgs set.

File: pkgs/tools/misc/timg/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  src = fetchFromGitHub {
+    owner = "hzeller";
+    repo = "timg";
+    rev = "4a300cfe9c159ffb5d357b41b3c69c57219da9e8";
+    sha256 = "0vx4i1ibgc0rmjdb607qbdbwxz9m89vam3hgzz2kvji1zl4b90ns";
+  };
+
+  nativeBuildInputs = with pkgs; [gnumake];
+  buildInputs = with pkgs; [libwebp graphicsmagick];
```
Comment (Lassulus) [inline pkgs/tools/misc/timg/default.nix:17]: the `with pkgs;` will be unnecessary if they are imported explicitly

File: pkgs/tools/misc/timg/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/tools/misc/timg/default.nix:1]: Please do not use pkgs as an input.

File: pkgs/tools/misc/timg/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{ stdenv, pkgs, fetchFromGitHub }:
+
+with stdenv.lib;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/timg/default.nix:3]: Please do not use with ove rthe entire package.

File: pkgs/tools/misc/timg/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  version = "2018-04-22";
+
+  src = fetchFromGitHub {
+    owner = "hzeller";
+    repo = "timg";
+    rev = "4a300cfe9c159ffb5d357b41b3c69c57219da9e8";
+    sha256 = "0vx4i1ibgc0rmjdb607qbdbwxz9m89vam3hgzz2kvji1zl4b90ns";
+  };
+
+  nativeBuildInputs = with pkgs; [gnumake];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/timg/default.nix:16]:
```suggestion
  nativeBuildInputs =[ gnumake ];
```

File: pkgs/tools/misc/timg/default.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  buildInputs = with pkgs; [libwebp graphicsmagick];
+
+  phases = [ "unpackPhase" "buildPhase" "installPhase" ];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/timg/default.nix:19]: Do not overwrite phases.

File: pkgs/tools/misc/timg/default.nix [hunk 6]
```diff
@@ -0,0 +1,34 @@
+{ stdenv, pkgs, fetchFromGitHub }:
+
+with stdenv.lib;
+
+stdenv.mkDerivation rec {
+  name = "timg-${version}";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/timg/default.nix:6]:
```suggestion
  pname = "timg";
```

File: pkgs/tools/misc/tiv/default.nix [hunk 7]
```diff
@@ -0,0 +1,32 @@
+{ stdenv, pkgs, fetchFromGitHub }:
+
+with stdenv.lib;
+
+stdenv.mkDerivation rec {
+  name = "tiv-${version}";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/tiv/default.nix:6]:
```suggestion
  pname = "tiv";
```

File: pkgs/tools/misc/tiv/default.nix [hunk 8]
```diff
@@ -0,0 +1,32 @@
+{ stdenv, pkgs, fetchFromGitHub }:
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/tiv/default.nix:1]: Basically all feedback from timg allpies here, too.

File: pkgs/tools/misc/tiv/default.nix [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  };
+
+  nativeBuildInputs = with pkgs; [gnumake gcc];
+  buildInputs = with pkgs; [imagemagick];
+  phases = ["unpackPhase" "buildPhase" "installPhase"];
+  buildPhase = ''make -C src/main/cpp'';
+  installPhase = ''
+    mkdir -p "$out/bin/"
+    cp -a src/main/cpp/tiv "$out/bin/"
+  '';
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/tiv/default.nix:23]:
```suggestion
  nativeBuildInputs = [ gcc gnumake ];

  buildInputs = [ imagemagick ];
  
  buildPhase = ''
    make -C src/main/cpp
  '';

  installPhase = ''
    mkdir -p "$out/bin/"
    cp -a src/main/cpp/tiv "$out/bin/"
  '';
```

File: pkgs/tools/misc/tiv/default.nix [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+  meta = {
+    license = licenses.asl20;
+    homepage = "https://github.com/stefanhaustein/TerminalImageViewer";
+    description = "Small C++ program to display images in a (modern) terminal using RGB ANSI codes and unicode 4x8 block graphic characters.";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/tiv/default.nix:28]:
```suggestion
    description = "Displays images in a (modern) terminal using RGB ANSI codes and unicode 4x8 block graphic characters";
```

File: pkgs/tools/misc/timg/default.nix [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+  buildPhase = ''make -C src'';
+  installPhase = ''
+    mkdir -p "$out/bin/"
+    cp -a src/timg "$out/bin/timg"
+  '';
+
+  meta = {
+    license = licenses.gpl2;
+    homepage = "https://github.com/hzeller/timg";
+    description = "A viewer that uses 24-Bit color capabilities and 1x2 unicode character blocks to display images in the terminal.";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/timg/default.nix:30]:
```suggestion
    description = "Viewer that uses 24-Bit color capabilities and 1x2 unicode character blocks to display images in the terminal";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please change the version numbers in your commits and PR title to the ones in the package definitions.

**Comment by ToxicFrog** [comment]
It has been A Year and I haven't had the bandwidth to go back and deal with my PRs, but I really should do that over the long weekend or something, Sorry.

**Comment by ToxicFrog** [comment]
Looks like timg and tiv have both been added to nixpkgs by other people since, so I'm dropping this.

**Pull request closed without merging**